lib/openstack_api/openstack_server.py

```diff
@@ -4,25 +4,31 @@
 from openstack.compute.v2.image import Image
 
 
-def migrate_server(
+def snapshot_and_migrate_server(
     conn: Connection,
     server_id: str,
+    server_status: str,
+    snapshot: bool,
     dest_host: Optional[str] = None,
-    live: bool = True,
 ) -> None:
     """
-    Migrate a server to a new host
+    Optionally snapshot a server and then migrate it to a new host
     :param conn: Openstack Connection
     :param server_id: Server ID to migrate
-    :param dest_host: Optional host to migrate to, otherwise choosen by scheduler
-    :param live: True for live migration
+    :param server_status: Status of machine to migrate - must be ACTIVE or SHUTOFF
+    :param dest_host: Optional host to migrate to, otherwise chosen by scheduler
     """
-    if not live:
-        conn.compute.migrate_server(server=server_id, host=dest_host)
-    else:
-        conn.compute.live_migrate_server(
-            server=server_id, host=dest_host, block_migration=True
+    if snapshot:
+        snapshot_server(conn=conn, server_id=server_id)
+    if server_status not in ["ACTIVE", "SHUTOFF"]:
+        raise ValueError(
+            f"Server status: {server_status}. The server must be ACTIVE or SHUTOFF to be migrated"
         )
```

Comment:
I wonder if it's better to check for active and shutoff before migrate? The way it currently is. We will build up snapshots of every VM even if we don't migrate them.
```suggestion
   if server_status not in ["ACTIVE", "SHUTOFF"]:
        raise ValueError(
            f"Server status: {server_status}. The server must be ACTIVE or SHUTOFF to be migrated"
        )
    if snapshot:
        snapshot_server(conn=conn, server_id=server_id)
    
```

Reply:
Yeah that seems like a more sensible option to me, thanks- I will change it now